app/sose2017/views.py

```diff
@@ -119,11 +122,30 @@ def __init__(self, **kwargs):
 
     submit = SubmitField()
 
+def handle_zugangspasswort():
+    if 'REGISTRATION_PASSWORD' not in current_app.config:
+        return None
+
+    if 'zugangspasswort' not in session:
+        form = RegistrationPasswordForm()
+        if form.validate_on_submit():
+            if form.passwort.data.lower() == current_app.config['REGISTRATION_PASSWORD'].lower():
+                session['zugangspasswort'] = True
+                return None
+            else:
+                form.passwort.data = ""
+                form.passwort.errors.append("Das Zugangspasswort sollte Dir in der Einladung zugeschickt worden sein.")
+        return form
+
```

Comment:
But anyway for completeness you should check the password if it is in the session